src/transaction/log_meta.hpp

```diff
@@ -50,6 +57,14 @@ namespace cublog
     private:
       using checkpoint_container_t = std::map<log_lsa, checkpoint_info>;
 
+    private:
+      /* flag parallel to 'log global header is shutdown':
+       *  - false: server has not been clean shut down
+       *  - true: has been clean shut down
+       * after start, flag is set to 'false' and the meta log is saved
+       */
+      bool m_is_tsrs_shutdown = false;
```

Comment:
Naming is not ideal. It is intended to be different than `log_Gl.hdr.is_shutdown`.